coderd/parameters.go

```diff
@@ -269,3 +275,31 @@ func (api *API) getWorkspaceOwnerData(
 		Groups:       groupNames,
 	}, nil
 }
+
+// parameterProvisionerVersionDiagnostic checks the version of the provisioner
+// used to create the template version. If the version is less than 1.5, it
+// returns a warning diagnostic. Only versions 1.5+ return the module & plan data
+// required.
+func parameterProvisionerVersionDiagnostic(tf database.TemplateVersionTerraformValue) hcl.Diagnostics {
+	missingMetaData := hcl.Diagnostic{
+		Severity: hcl.DiagError,
+		Summary:  "This template version is missing required metadata to support dynamic parameters. Go back to the classic creation flow.",
+		Detail:   "To restore full functionality, please re-import the terraform as a new template version.",
+	}
+
+	if tf.ProvisionerdVersion == "" {
+		return hcl.Diagnostics{&missingMetaData}
+	}
+
+	major, minor, err := apiversion.Parse(tf.ProvisionerdVersion)
+	if err != nil || tf.ProvisionerdVersion == "" {
+		return hcl.Diagnostics{&missingMetaData}
+	} else if major < 1 || (major == 1 && minor < 5) {
+		missingMetaData.Detail = "This template version requires provisioner v1.5 or newer to support dynamic parameters. " +
```

Comment:
Well, it's not really about the template version requiring it, the experiment is enabled at a deployment level. Really what's happening is that the template version doesn't properly _support_ dynamic parameters.

```suggestion
		missingMetadata.Detail = "This template version does not support dynamic parameters. " +
```